src/runtime/program/vote/execute.zig

```diff
@@ -138,10 +174,287 @@ fn intializeAccount(
+            return InstructionError.Custom;
+        };
+    }
+
+    try authorize(
+        allocator,
+        ic,
+        vote_account,
+        new_authority,
+        authorization_type,
+        clock,
+        if (expected_authority_keys.count() > 0)
+            expected_authority_keys
+        else
+            null,
+    );
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/0603d1cbc3ac6737df8c9e587c1b7a5c870e90f4/programs/vote/src/vote_processor.rs#L96-L102
+fn executeAuthorizeCheckedWithSeed(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    authorization_type: VoteAuthorize,
+    current_authority_derived_key_owner: Pubkey,
+    current_authority_derived_key_seed: []const u8,
+) InstructionError!void {
+    try ic.info.checkNumberOfAccounts(4);
+
+    var new_authority = try ic.borrowInstructionAccount(
+        @intFromEnum(vote_instruction.VoteAuthorizeCheckedWithSeedArgs.AccountIndex.new_authority),
+    );
+    defer new_authority.release();
+
+    try authorizeWithSeed(
+        allocator,
+        ic,
+        vote_account,
+        new_authority.pubkey,
+        authorization_type,
+        current_authority_derived_key_owner,
+        current_authority_derived_key_seed,
+        @intFromEnum(vote_instruction.VoteAuthorizeCheckedWithSeedArgs.AccountIndex.base_key),
+        @intFromEnum(vote_instruction.VoteAuthorizeCheckedWithSeedArgs.AccountIndex.clock_sysvar),
+    );
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/0603d1cbc3ac6737df8c9e587c1b7a5c870e90f4/programs/vote/src/vote_processor.rs#L239-L248
+fn executeAuthorizeChecked(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    vote_authorize: vote_instruction.VoteAuthorize,
+) InstructionError!void {
+    try ic.info.checkNumberOfAccounts(4);
+
+    var new_authority = try ic.borrowInstructionAccount(
+        @intFromEnum(vote_instruction.VoteAuthorize.AccountIndex.new_signer),
+    );
+    defer new_authority.release();
+
+    if (!ic.info.isPubkeySigner(new_authority.pubkey)) {
+        return InstructionError.MissingRequiredSignature;
+    }
+
+    const clock = try ic.getSysvarWithAccountCheck(
+        Clock,
+        @intFromEnum(vote_instruction.VoteAuthorize.AccountIndex.clock_sysvar),
+    );
+
+    const autorize = switch (vote_authorize) {
+        .Voter => VoteAuthorize.voter,
+        .Withdrawer => VoteAuthorize.withdrawer,
+    };
+
+    try authorize(
+        allocator,
+        ic,
+        vote_account,
+        new_authority.pubkey,
+        autorize,
+        clock,
+        null,
+    );
+}
+
+// TODO: Move this to instruction_context.zig
+fn verifyAuthorizedSigner(
+    authorized: Pubkey,
+    signers: std.AutoHashMap(Pubkey, void),
+) InstructionError!bool {
+    if (signers.contains(authorized)) {
+        return true;
+    } else {
+        return InstructionError.MissingRequiredSignature;
+    }
 }
```

Comment:
should we move this now?

Reply:
Started doing that, but found out that it does not take `InstructionInfo` as self (like the other related methods: `isPubkeySigner` etc), and both of it's inputs are not tied to the instruction. So feels more like a utility function.

I have left it here, and removed the todo f804999 if this is needed somewhere else perhaps then it can be moved.